### DISCUSSION TITLE
How does "super" compute its alignment score?

### DISCUSSION THREAD

**Discussion by danieljnilson** [schrodinger/pymol-open-source#406] [Q&A] [answered]
As the title says.

I am writing a summary in my thesis of a pipeline I wrote that uses the PyMOL api. I'd like to be able to detail how the score is calculated as I make reference/use of it multiple times.

**Comment by ye11owSub** [comment]
hi @danieljnilson 
Is [this](https://github.com/schrodinger/pymol-open-source/blob/master/modules/pymol/fitting.py#L306-L394) the code you were looking for?

**Comment by danieljnilson** [comment]
Hi @ye11owSub ! 
Thank you for your response. That is a good find; though it appears to be more of like the function `super` description. It does not seem to describe the generated output. I will try to use this information to dig more later.

**Reply by ye11owSub** [reply]
This function is written in C (Python is just an API for it), so it would be more helpful to look at [this source code](https://github.com/schrodinger/pymol-open-source/blob/master/layer3/Executive.cpp#L7598-L7714)

**Comment by JarrettSJohnson** [comment]
The PyMOLWiki entry for `super` gives basic information https://pymolwiki.org/index.php/Super , and the C/C++ code is unfortunately not very clear to read. The major difference between this and the other `align` function, is that `super` first performs a _sequence-independent_ residue pairing determined by local dihedral information (dihedral angles from CA-CB pairs along with other dihedrals against other atoms within a given radius (default 12 angstroms). Dynamic programming is used to match residue pairs by their local structural similarity; and from there, I believe you get your typical RMSD refinement stages just as you do with `align`.

Feel free to ask any questions if you need further clarification.

**Reply by danieljnilson** [reply]
Hello @JarrettSJohnson  . Thank you for your response and the url link. I am familiar with that pymol wiki on `super` ; it is a very helpful page describing a high level overview of how that function works. Unfortunately, it does not describe how the "MatchAlign: score" is determined for each use of the `super` .  I do not see a description in the wiki page for 'align' either. I will keep looking in the source code too. Thank you for your assistance.

**Reply by JarrettSJohnson** [reply, marked as answer]
From what I understand, the `MatchAlign` score takes an arbitrary scoring matrix along with dynamic programming parameters (gap penalties, etc...) to discover the optimal alignment via something similar to `Needleman-Wunsch`. For `super`, the scoring matrix is as I mentioned above, information regarding its local structural information (mostly dihedrals). While `align` will generate the scoring matrix based off of one like the `BLOSUM62`.

**Reply by danieljnilson** [reply]
Thank you. If there is a paper or some sort of reference to go with that, it would be great. However, for now this gives me something to work with.
